lib/cosmos/tools/tlm_grapher/tabbed_plots_tool/tabbed_plots_realtime_thread.rb

```diff
@@ -65,6 +65,12 @@ def kill
 
     def graceful_kill
       # Just to remove warning
+      if defined? Qt and Thread.current == Thread.main
+        5.times do
```

Comment:
I'll add a note.  Its so the realtime thread can die gracefully not the whole gui.  The problem is things get stuck on Qt.execute_in_main_thread and so you need to call processEvents so they can get unstuck.